TUTORIAL.md

```diff
@@ -0,0 +1,371 @@
+
+  ```shell
+  export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
+  ```
+
+* Start Airflow by running `airflow standalone` command and check out the Webserver
+at [`http://localhost:8080/`](http://localhost:8080/).
+
+## Setup Airflow connections
+
+* In your Astronomer Airflow UI select Admin-->Connectors
+
+![connections](docs/images/airflow-webserver-connection.png)
+
+
+### Connect S3 to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+Set these fields:
+
+* Connection Id: `aws_default`
+* Connection Type: `S3`
+* Extra: `{"aws_access_key_id": "<your_access_key>", "aws_secret_access_key": "<you_secret_access_key>"}`
+
+### Connect Snowflake to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+* Connection Id: `snowflake_default`
+* Connection Type: `Snowflake`
+* Host: `https://<account>.<region>.snowflakecomputing.com/`. This is the URL where you can log into your Snowflake account
+* Schema:
+* Login:
+* Password:
+* Account:
+* Database:
+* Region: (something like us-east-1 or us-central1.gcp)
+* Role:
+* Warehouse:
+
+
+***
+# Create and populate some tables in Snowflake
+
+We'll create some auxiliary tables in Snowflake and populate with a small amount of data for our ETL example.
+
+* Create and populate a customer table to join with an orders table that we create with the Astro Python SDK:
+
+```sql
+
+CREATE OR REPLACE TABLE customers_table (customer_id CHAR(10), customer_name VARCHAR(100), type VARCHAR(10) )
+
+INSERT INTO customers_table (CUSTOMER_ID, CUSTOMER_NAME,TYPE) VALUES ('CUST1','NAME1','TYPE1'),('CUST2','NAME2','TYPE1'),('CUST3','NAME3','TYPE2')
+```
+
+* Create and populate a reporting table into which we'll merge our transformed data:
+
+```sql
+
+CREATE OR REPLACE TABLE reporting_table (
+    CUSTOMER_ID CHAR(10), CUSTOMER_NAME VARCHAR(100), ORDER_ID CHAR(10), PURCHASE_DATE DATE, AMOUNT FLOAT, TYPE CHAR(10));
+
+INSERT INTO reporting_table (CUSTOMER_ID, CUSTOMER_NAME, ORDER_ID, PURCHASE_DATE, AMOUNT, TYPE) VALUES
+('INCORRECT_CUSTOMER_ID','INCORRECT_CUSTOMER_NAME','ORDER2','2/2/2022',200,'TYPE1'),
+('CUST3','NAME3','ORDER3','3/3/2023',300,'TYPE2'),
+('CUST4','NAME4','ORDER4','4/4/2022',400,'TYPE2')
+```
+***
+# Simple ETL workflow
+
+Use your favorite code editor or text editor to copy-paste the following code into a *.py file in <your-project>/dags/ directory.
+
+Here's the code for the simple ETL workflow:
+
+```python
+from datetime import datetime
+
+from airflow.models import DAG
+from astro.sql.table import Table, Metadata
+from astro.sql import dataframe as df
+from astro.files import File
+from astro import sql as aql
+from pandas import DataFrame
+
+S3_FILE_PATH = "s3://<aws-bucket-name>"
+S3_CONN_ID = "aws_default"
+SNOWFLAKE_CONN_ID = "snowflake_default"
+SNOWFLAKE_ORDERS = "orders_table"
+SNOWFLAKE_FILTERED_ORDERS = "filtered_table"
+SNOWFLAKE_DATABASE = "ASTRO_SDK_DB"
+SNOWFLAKE_SCHEMA = "ASTRO_SDK_SCHEMA"
+SNOWFLAKE_WAREHOUSE = "ASTRO_SDK_DW"
```

Comment:
These properties could be included only in the connection, simplifying the example